File: lib/java/se/lth/control/labcomm2006/Decoder.java
package se.lth.control.labcomm2006;

import java.io.IOException;

public interface Decoder {
  
  public void register(SampleDispatcher dispatcher, 
		       SampleHandler handler) throws IOException;
  public boolean decodeBoolean() throws IOException;
  public byte decodeByte() throws IOException;
  public short decodeShort() throws IOException;
  public int decodeInt() throws IOException;
  public long decodeLong() throws IOException;
  public float decodeFloat() throws IOException;
  public double decodeDouble() throws IOException;
  public String decodeString() throws IOException;

}

File: lib/csharp/se/lth/control/labcomm2006/Decoder.cs
using System;

namespace se.lth.control.labcomm2006 {

  public interface Decoder {

    void register(SampleDispatcher dispatcher, 
		  SampleHandler handler);

    bool decodeBoolean();
    byte decodeByte();
    short decodeShort();
    int decodeInt();
    long decodeLong();
    float decodeFloat();
    double decodeDouble();
    String decodeString();

  }

} 
